articles/problem-framing/index.md

```diff
@@ -0,0 +1,98 @@
+
+On the other hand, machine learning may prove to be the most feasible approach. The organization may have enough resources to implement and maintain a machine learning solution. For instance, the organization may be capable of hiring a team of data scientists. In such a case, a strategy would have to be developed around the proposed approach. This would first involve formulating a machine learning question.
+
+#### Formulating a machine learning question
+
+At this point, we have already identified and defined the business problem. To formulate a machine learning question, we first need to understand the problem domain. In this case, we seek to classify customers into their correct segments based on their buying habits.
+
+Relating the problem to machine learning would result in understanding the algorithm that could provide the best value. We would have to ask questions such as; could it be best framed as a classification problem? Is it a regression problem? Correct analysis of the problem shows that it is a classification problem. As a result, a classification algorithm would be an obvious choice.
+
+The next step would involve understanding how the output of the proposed classification algorithm would impact decision making. The solution would have to fit into the overall picture of the organization. For instance, the classification of customers would potentially allow the organization to develop better products. It could allow the production of more personalized products. It could also result in a better understanding of how to produce the right number of products for given customer segments to allow better inventory clearance. The model’s output data would help make all these decisions.
+
+We would then need to identify the sources of data for our model. Data powers machine learning algorithms. The data that would offer the best value would have to be of the utmost quality. Data of lesser quality would undermine the goals of our model.
+
+#### Beginning with a simple model
+
+To allow for a not only explainable but interpretable model, one may consider using a simple model at this stage. A simple model may be described as one with the least possible complexity required to correctly carry out a task. The use of such a model allows for controllable debugging. Simple models also provide a good baseline. They can help evaluate whether there is a need for a more complex model. Complex models provide less explainability. They are harder and take longer to train compared to simple ones. If the need for more complexity is justified, it may be implemented later on.
+
+### Wrapping up
+
+Problem framing is a key step in the machine learning workflow. Since it’s arguably less technical than other stages in a machine learning workflow, it might be inaccurately perceived as an easy step. Yet, the challenges we covered show otherwise. The best practices we have covered should be of help when framing machine learning problems.
```

Comment:
You mention problem framing and then data analysis. So does data collection and understanding sources of data come under problem framing as well? Can you elaborate on some of the issues under data collection?